clients/src/main/java/org/apache/kafka/common/metrics/MetricConfig.java

```diff
@@ -65,6 +69,15 @@ public MetricConfig timeWindow(long window, TimeUnit unit) {
         return this;
     }
 
+    public Map<String, String> tags() {
```

Comment:
Yes it is a good point and should remove quite some lines of code. But this will touch a lot of files and potentially cause conflict with other commits. Let me give it a shot.